### PR TITLE
Improve timeout reporting

### PR DESCRIPTION
**Pull request by reset**
This introduces three changes which will improve message routing error reporting - specifically, timeout messages will be massaged into the appropriate net error. We also now map Timeout net errors to a more appropriate HTTP status code (504 instead of 408)

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/net/src/oauth/github.rs [hunk 1]
```diff
@@ -29,6 +30,7 @@ use config;
 use error::{Error, Result};
 
 const USER_AGENT: &'static str = "Habitat-Builder";
+const HTTP_TIMEOUT: u64 = 3_000;
```
Comment (smurawski) [inline components/net/src/oauth/github.rs]: Should this be a value that we read from a configuration file?  Thinking of a private depot behind proxies, though 3 seconds does seem like a reasonably long time.

**Comment by smurawski** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/170930396).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/170930396).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**